app/views/prior_authority/applications/search.html.erb

```diff
@@ -39,9 +39,10 @@
         </thead>
         <tbody class="govuk-table__body app-task-list__items">
           <% @model.each do |application| %>
+            <% @link = application.state == 'draft' ? prior_authority_steps_start_page_path(application) : prior_authority_application_path(application) %>
```

Comment:
Nits:

- as this is in an `each` loop you don't really want to be setting an ivar here. `link` would be better than `@link` as otherwise once you exit the loop you'll have a random value sitting around that could mess us up if elsewhere in the view or controller we want an actual `@link` ivar in future.

- It's slightly less error-prone to use `application.draft?` instead of `application.state == 'draft'` - shorter and will blow up if there are typos

- I think the best fix would be to handle the situation where somehow a user tries to visit `prior_authority_application_path` for a draft application. Doing what NSM does and gracefully redirecting means that if we ever have another situation where we accidentally link to the wrong place it won't blow up again.

- Agree with Ivan that tests would be valuable.